serde_codegen/src/ser.rs

```diff
@@ -586,7 +525,6 @@ fn serialize_struct_variant(
         .with_fields(
             fields.iter().map(|field| {
                 builder.struct_field(field.ident.expect("struct has unnamed fields"))
-                    .with_attrs(field.attrs.iter().cloned())
```

Comment:
To be honest I can't remember why I did this. This might have been from some attempt to unify variant structs with regular structs in order to generically handle fields, but if removing this still passes all the unit tests then it's probably not needed, so I think it'll be okay to remove it. And if it does break things, then shame on me for not writing a unit test for this case :)